### DISCUSSION TITLE
Support for vanilla javascript

### DISCUSSION THREAD

**Discussion by tusharnaresh** [TanStack/virtual#455] [Q&A] [answered]
I was trying to use virtual in core js project but i am unable to make it work. Even after providing a valid scrollable element in getScrollElement function, the scrollToFn function never triggers and scrollElement inside VIrtualizer instance is null. 
Here is the link to my implementation
https://stackblitz.com/edit/js-ur5mst?file=index.js
Any help would be much appreciated.

**Comment by piecyk** [comment, marked as answer]
Hi, you also need to provide observeElementRect, observeElementOffset, then after instance is created call `instance._willUpdate()` it will call provided options and setup the virtualizer. Something like https://stackblitz.com/edit/js-yc7vby?file=index.js,style.css,index.html

**Comment by tusharnaresh** [comment]
There seems to be one issue in the above implementation, I observed that in this case the number of elements in scroll container do not get limited. Even though I am viewing 5 elements on the container at a time i can see thousands of elements in my html elements.
Here is the link for the implementation:
https://stackblitz.com/edit/js-ur5mst?file=index.js
![Capture](https://user-images.githubusercontent.com/89134825/209633970-8c916d6e-bbae-4aca-95c6-7644abc024d6.JPG)

**Reply by piecyk** [reply]
It's because you don't remove items that are no longer visible.

**Reply by tusharnaresh** [reply]
got it. Thanks for the help.
